src/manager.js

```diff
@@ -305,50 +320,99 @@ AdManager.prototype.setSlotTargeting = function(element, slot) {
+
+  var slotName = this.options.dfpSiteCode || "",
+    kinjaMeta = window.kinja.meta,
+    postMeta = window.kinja.postMeta || {},
+    currentBlog = kinjaMeta.blog,
+    blogSales = kinjaMeta.blogSales,
+    pageType = kinjaMeta.pageType,
+    getExperimentVariation = function(scope) {
+      var expScope = scope || window,
+        variation = expScope.cxApi ? expScope.cxApi.getChosenVariation(expScope.gaExperimentId) : null;
+      return (variation !== null && variation >= 0 && variation < 26) ? String.fromCharCode(variation + 65) : null;
+    },
+
+    getExperimentId = function(scope) {
+      var expScope = scope || window;
+      return (expScope.gaExperimentId !== undefined) ? expScope.gaExperimentId : null;
+    },
+
+    getQueryParameter = function(name) {
+      var regexS = '[\\?&]' + name + '=([^&#]*)',
+        regex = new RegExp(regexS),
+        results = regex.exec(window.location.search),
+        retval = '';
+
+      if (results) {
+        return decodeURIComponent(results[1].replace(/\+/g, ' '));
+      }
+
+      return retval;
+    },
+    expVar = (getExperimentVariation() !== null && getExperimentId() !== null) ? getExperimentId() + '_' + getExperimentVariation() : null,
+    paramZone = getQueryParameter('adzone'),
+    forcedAdZone = (function(pZone, meta) {
+      var tags = meta.tags,
+        forceNonCollapse = /why your team sucks|wyts/.test(tags),
+        forceCollapse = /nsfw|gamergate/.test(tags),
+        forceCollapseZone = !forceNonCollapse && forceCollapse ? 'collapse' : null,
+        post = meta.post,
+        postZone = post ? post.adZone : null;
+      return pZone || forceCollapseZone || postZone;
+      // kinjaMeta.post is not available at this time so we must reference kinja.postMeta
+    }(paramZone, postMeta)),
+
+    // figure out the defined adZone
+    adZone = forcedAdZone && forcedAdZone === 'collapse' ? 'collapse' : (pageType === 'frontpage' ? 'front' : pageType),
+    networkId = blogSales.adNetworkId || 4246,
+    unitName = '/' + [networkId, blogSales.adSiteName, adZone].join('/'),
+    adUnitConfig = this.adUnits.units[element.dataset.adUnit],
+    size = adUnitConfig.sizes[0][1],
+    tags = kinjaMeta.tags,
+    post = postMeta.post;
+
+  element.id = this.generateId();
+  slot = this.googletag.defineSlot(unitName, size, element.id);
+
   if (element.id && element.id in this.slots) {
     // Slot has already been configured
     return this.slots[element.id];
   }
 
-  element.id = this.generateId();
-
   if (!element.dataset) {
     this.logMessage('Browser does not support dataset', ERROR);
     return;
   }
 
-  var adUnitConfig = this.adUnits.units[element.dataset.adUnit];
   if (!adUnitConfig) {
     this.logMessage('Ad unit (' + element.dataset.adUnit + ') missing configuration', ERROR);
     return;
   }
 
-  var positionTargeting = adUnitConfig.slotName || element.dataset.adUnit;
-
-  var slotName = this.options.dfpSiteCode;
-  if (window.dfpSiteSection) {
```

Comment:
@spra85 I think i got this right in the latest changeset.  Let me know what you think.